clients/drcachesim/scheduler/scheduler.h

```diff
@@ -1033,6 +1048,10 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
     next_record(output_ordinal_t output, RecordType &record, input_info_t *&input,
                 uint64_t cur_time = 0);
 
+    // Undoes the last read.  May only be called once between next_record() calls.
```

Comment:
As mentioned above, please clarify how this interacts with speculation, including speculation that uses the queue_next_record flag.

Reply:
Probably worth adding a test for this interaction as well.

Reply:
Done (disallowed to mix), and done on adding a test of each failure case.